python-checks/src/main/java/org/sonar/python/checks/DateTimeNoPyTzTimeZoneInConstructor.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks;
+
+import java.util.List;
+import java.util.Objects;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.ArgList;
+import org.sonar.plugins.python.api.tree.Argument;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.cfg.fixpoint.ReachingDefinitionsAnalysis;
+
+@Rule(key = "S6887")
+public class DateTimeNoPyTzTimeZoneInConstructor extends PythonSubscriptionCheck {
+
+  private ReachingDefinitionsAnalysis reachingDefinitionsAnalysis;
+  private static final String MESSAGE = "pytz.timezone should not be passed to the datetime.datetime constructor";
```

Comment:
This is my bad, as issue messages should be part of the rule specification and I forgot to add it for this one.
In any case, we have some [guidelines](https://docs.sonarsource.com/sonarqube/latest/extension-guide/adding-coding-rules/) to write issue messages. One of them is that the issue message should generally be imperative.

I'd suggest something like `Don't pass a "pytz.timezone" here.` or `Don't pass a "pytz.timezone" to the "datetime.datetime" constructor.`